social-tw/social-tw-website
# Feature/login frontend 0629 add lib and UI

## Summary

This PR includes the successful installation and configuration of Tailwind CSS & DaisyUI and the completion of the Landing page. I have also introduced new components including Text, Button, and Post. The PR also includes an unsuccessful attempt at configuring twitter-react-login. Note that the current state does not include RWD implementation, and no functional operations have been realized yet.

## Linked Issue

Not applicable.

## Details

- Installed and configured Tailwind CSS & DaisyUI.
- Completed the Landing Page.
- Introduced Text, Button, and Post components.
- Attempted (unsuccessfully) to configure twitter-react-login.
- Did not implement RWD or any functional operations.

## Impacted Areas

The main areas impacted include:

Setup files for Tailwind CSS & DaisyUI.
Landing page files.
New component files for Text, Button, and Post.
Config files for twitter-react-login.

## Tests

Currently, no specific tests were performed as the changes mostly include installations and UI creation without functional operations. As we progress, unit and integration tests will be written to verify the functionality.

## Possible Impacts

This PR shouldn't affect any existing features as most of the changes are additive. However, it's important to keep an eye on any potential conflicts between the newly added dependencies and the existing ones.

## Visual Materials

![截圖 2023-06-29 下午8 32 41](https://github.com/social-tw/social-tw-website/assets/5743898/596f0234-c5ce-474b-9d3d-a0d57ad877c6)
![截圖 2023-06-29 下午8 32 48](https://github.com/social-tw/social-tw-website/assets/5743898/17e175e1-897f-4e8e-a3d7-84f055ac8fe0)

## Verification Steps

To verify the changes:

Install the project dependencies.
Run the project.
Inspect the Landing page and the new components.

## Todo

Here is the breakdown of what is left to be done:

- [ ]  Fix the configuration of twitter-react-login.
- [ ]  Implement RWD.
- [ ]  Add functional operations to the components.
- [ ]  Test the changes.

## Checklist

- [x]  Implement Landing page ui from figma
